src/rust/engine/src/nodes.rs

```diff
@@ -915,7 +923,7 @@ impl NodeKey {
         keystr(&s.subject),
         typstr(&s.product)
       ),
-      &NodeKey::Snapshot(ref s) => format!("Snapshot({})", keystr(&s.0)),
```

Comment:
As above with `impl Debug for Key`: we'll do a widespread cleanup of error messages at some point: would leave this as is.

Reply:
I reverted this change -- thanks for pointing it out (wasn't really thought through on my part).